app/notifications/process_notifications.py

```diff
@@ -276,16 +264,16 @@ def send_to_queue_for_recipient_info_based_on_recipient_identifier(
                 queue=QueueNames.SEND_ONSITE_NOTIFICATION
             ),
             lookup_contact_info.si(notification.id).set(queue=QueueNames.LOOKUP_CONTACT_INFO),
-            deliver_task.si(notification.id).set(queue=deliver_queue),
         ]
 
-        if is_feature_enabled(FeatureFlag.CHECK_RECIPIENT_COMMUNICATION_PERMISSIONS_ENABLED) and communication_item_id:
-            tasks.insert(
-                len(tasks) - 1,
-                lookup_recipient_communication_permissions.si(notification.id).set(
-                    queue=QueueNames.COMMUNICATION_ITEM_PERMISSIONS
-                ),
-            )
+    if communication_item_id:
+        tasks.append(
+            lookup_recipient_communication_permissions.si(notification.id).set(
+                queue=QueueNames.COMMUNICATION_ITEM_PERMISSIONS
+            ),
+        )
+
+    tasks.append(deliver_task.si(notification.id).set(queue=deliver_queue))
```

Comment:
Just a bit of readability cleanup here; I don't like the "insert before the last element in the list" pattern, and there was some duplication anyway.  Two birds, one stone.